src/main/sensors/battery.c

```diff
@@ -355,40 +358,67 @@ void powerMeterUpdate(timeUs_t timeDelta)
     mWhDrawn = mWhDrawnRaw / (3600 * 100);
 }
 
-void sagCompensatedVBatUpdate(timeUs_t currentTime)
+void sagCompensatedVBatUpdate(timeUs_t currentTime, timeUs_t timeDelta)
 {
     static timeUs_t recordTimestamp = 0;
     static int32_t amperageRecord;
     static uint16_t vbatRecord;
+    static uint8_t impedanceSampleCount = 0;
+    static pt1Filter_t impedanceFilterState = PT1_FILTER_INITIALIZER;
```

Comment:
Initializing filter with RC = 0 doesn't make much sense. I don't see where in the code filter is properly initialized with cutoff frequency.

Reply:
It is done line 401
`sagCompVBatFilterState.RC = sagCompensatedVBatSample < sagCompVBatFilterState.state ? 40 : 500;`

I will add functions to access the time constant and last value.